### DISCUSSION TITLE
Draw a smooth arc from 12 'o clock

### DISCUSSION THREAD

**Discussion by bbuster2** [Bodmer/TFT_eSPI#2549] [Q&A - General]
Is there a way to draw a smooth arc from the top?

Since it starts from the bottom and can only range from 0 to 360 i have to start at 180 till 360 and after that create a new arc from 0 till 180.

With the anti-aliasing it shows a "border" at the bottom.
Is it possible to rotate just the arc by 180 degrees?

**Comment by Bodmer** [comment]
Start at 180 and end at any angle in range 0-360. 180 to 179 will be the largest arc angle. For the circle case draw 0 to 360.

**Comment by skiddykiddie** [comment]
Thanks so much for your great work Bodmer.  Im also trying to figure this out.

Ive got a smootharc that grows from about 231 deg to 0, then from 360 to 114 (from left to rignt along the bottom of the display).  Using the same driving input for this continual arc is a real nightmare.  If its possible to relocate the zero point that helps a ton.

Ive been fighting trying to make code that looks at the integer and draws the arc to the zero line then continues on, but its just like pushing a boulder uphill, any little change and it falls apart.

**Reply by Bodmer** [reply]
drawSmoothArc will draw from say 231 to 114 with a single call. The code that handles this is in the library.

drawArc will require that arc to be drawn in 2 segments.

**Comment by skiddykiddie** [comment]
I can do that, and its fine for a static arc, but an arc that is changing its size (value) is a bit harder.

It appears that a smoothArc cant be zero in span (thickness).  So for a dynamic arc that draws out with a changing number (say starting at 50 to (49 - 0), and needs to cross over the zero line, the 2nd half of the arc will always have at least a 1 deg arc drawn at the zero line to work.  Continuing the arc after zero is ok (360 to (359 - 350).  So when the 2nd half comes down that messes up the continuity of the complete arc.

Seems like any arc that falls to a zero thickness will draw out a 360 degree full arc.

If that makes any sense. I tried to make it as clear as possible. 

Ill have to create extra code that removes the 2nd arc completely when not crossing the zero line, and then initiates the arc when crossing the zero line.  Its a bit of a mess to get this done.

![Screenshot 2023-04-14 103320](https://user-images.githubusercontent.com/130083791/232116834-81109b9f-4263-4ada-8213-754c9fd2b0c7.jpg)

**Reply by Bodmer** [reply]
Would it help if the "zero" position could be shifted to say the 12 o'clock position, or would that simply move the problem to another point on the circumference because you want 360 rotation of a small arc segment?

Can you post a complete sketch that demonstrates the problem so I can see what you are trying to achieve.

**Reply by skiddykiddie** [reply]
Shifted to 3 o'clock would be the ticket in this specific instance.  But I read you said that just getting this arc to work the way it does is a lot of work, so no biggie if its not setup to do what Im wanting to do.  The abilities built in are already awesome.  There are other ways Im sure, but the smoothArc that gets drawn looks really great, and just drawing the graphics without antialiasing definitely wouldnt look as good.

My sketch is a bit of a mess right now with many commented out lines to keep track of what changes Ive made.  Ill clean up a version and post it later.
